### ISSUE TITLE
[SEINE][pdx201] build fails with some lacking dependencies

### ISSUE THREAD

**Issue by AngryGami** [sonyxperiadev/bug_tracker#604]
**Platform:** Seine 
**Device:** pdx201
**Kernel version:** msm-4.14
**Android version:** 10

**Previously working on**
build for ganges/kirin seems to work fine

**Description**
My first attempt to build this image failed because there is no yet pre-built kernel for it so I had to build it myself. Though even after I got kernel build still fails with error:
```
FAILED: ninja: 'hardware/qcom/audio/hal/sm6125/platform.c', needed by 'out/target/product/pdx201/obj_arm/SHARED_LIBRARIES/audio.primary.sm6125_intermediates/sm6125/platform.o', missing and no known rule to make it
15:11:39 ninja failed with: exit status 1
```

**How to reproduce**
Here is how I've init my repo:

```
mkdir aosp10-build
cd aosp10-build/
repo init -u https://android.googlesource.com/platform/manifest -b android-10.0.0_r37
cd .repo/
git clone https://github.com/sonyxperiadev/local_manifests
cd local_manifests/
git checkout android-10.0.0_r37
cd ../..
repo sync
./repo_update.sh
```

I've changed `build-kernels-clang.sh` script so it includes seine platform:
```
-export MKDTIMG=$ANDROID_ROOT/out/host/linux-x86/bin/mkdtimg
+export MKDTIMG=$ANDROID_ROOT/prebuilts/misc/linux-x86/libufdt/mkdtimg
...
+SEINE="pdx201"
...
-PLATFORMS="yoshino nile ganges tama kumano"
+PLATFORMS="seine"
```
After that I've point my environment to proper prebuild clang:
```
export PATH=$PATH:/aosp2/aosp10-build/prebuilts/clang/host/linux-x86/clang-r353983d/bin/
```
Then I've ran kernel build:
```
cd kernel/sony/msm-4.14/common-kernel/
./build-kernels-clang.sh
```
Verified that new kernel file was added:
```
:/aosp2/aosp10-build/kernel/sony/msm-4.14/common-kernel$ ls kernel-dtb-pdx201
kernel-dtb-pdx201
```
Returned to build root, choose my lunch and started build:
```
cd /aosp2/aosp10-build/
. build/envsetup.sh
lunch 58
make -j4
```
And here is full build output:
```
============================================
PLATFORM_VERSION_CODENAME=REL
PLATFORM_VERSION=10
TARGET_PRODUCT=aosp_xqau52
TARGET_BUILD_VARIANT=userdebug
TARGET_BUILD_TYPE=release
TARGET_ARCH=arm64
TARGET_ARCH_VARIANT=armv8-a
TARGET_CPU_VARIANT=cortex-a73
TARGET_2ND_ARCH=arm
TARGET_2ND_ARCH_VARIANT=armv8-a
TARGET_2ND_CPU_VARIANT=cortex-a73
HOST_ARCH=x86_64
HOST_2ND_ARCH=x86
HOST_OS=linux
HOST_OS_EXTRA=Linux-4.4.0-119-generic-x86_64-Ubuntu-16.04.6-LTS
HOST_CROSS_OS=windows
HOST_CROSS_ARCH=x86
HOST_CROSS_2ND_ARCH=x86_64
HOST_BUILD_TYPE=release
BUILD_ID=QQ3A.200605.001
OUT_DIR=out
PRODUCT_SOONG_NAMESPACES=device/sony/seine vendor/qcom/opensource/core-utils
============================================
[ 99% 11829/11830] finishing build rules ...
platform_testing/build/tasks/tests/instrumentation_test_list.mk: warning: continuous_instrumentation_tests: Unknown installed file for module 'NexusLauncherOutOfProcTests'
platform_testing/build/tasks/tests/instrumentation_test_list.mk: warning: continuous_instrumentation_tests: Unknown installed file for module 'NexusLauncherDebug'
platform_testing/build/tasks/tests/instrumentation_test_list.mk: warning: continuous_instrumentation_tests: Unknown installed file for module 'NexusLauncherTests'
platform_testing/build/tasks/tests/platform_test_list.mk: warning: platform_tests: Unknown installed file for module 'LauncherRotationStressTest'
platform_testing/build/tasks/tests/platform_test_list.mk: warning: platform_tests: Unknown installed file for module 'PlatformScenarioTests'
FAILED: ninja: 'hardware/qcom/audio/hal/sm6125/platform.c', needed by 'out/target/product/pdx201/obj_arm/SHARED_LIBRARIES/audio.primary.sm6125_intermediates/sm6125/platform.o', missing and no known rule to make it
15:11:39 ninja failed with: exit status 1

#### failed to build some targets (02:53 (mm:ss)) ####
```

**Comment by MarijnS95** [comment]
Note that Seine has a dtbo:

```diff
+        seine)
+            DEVICE=$SEINE;
+            DTBO="true";;
```

Audio is still in the process of being reworked, but is almost there (now waiting on other devices to finalize the migration). Follow recent PRs across this project to learn more.

**Comment by AngryGami** [comment]
How it happens that I can buy Xperia 10 II device in the shop and it will work and at same time there is no aosp image for it? I mean all problems with this device already solved elsewhere by Sony how these solutions are not available to community?

**Comment by MarijnS95** [comment]
How it happens that no-one really complained when Kumano was unusable and unfinished for months, whereas a lot of effort was put in getting the **COMMUNITY DRIVEN** opensource variant for PDX201 ready by the time the device is released (minus audio, which you can just disable or merge all outstanding PRs) and we get complaints for it not being done yet?
Again, look around at the PRs created _as we speak_ by @kholk to finalize it all.

If you want to use the solutions applied in stock... use stock? A question that unfortunately doesn't sound friendly but I have no other way of putting it: do you know what this project is about? How it works? How it is built and by whom?

**Comment by AngryGami** [comment]
:)
> do you know what this project is about?
> How it works? 
> How it is built and by whom?
No, no and no...
I assumed that this entire effort is coming from Sony and therefore they backing this project in one way or another. They do produce consumer grade devices and obviously also produce OS build for these. I surprised that you (I mean people that work on this project) have to reinvent the wheel here. If you have time and/or desire to explain me why this is the way it is - I'm all ears.

**Comment by leeoniya** [comment]
> If you have time and/or desire to explain me why this is the way it is - I'm all ears.

i'd be interested to understand this as well. the exact nature of the Sony <> SODP project has always been unclear. all i have heard is that you guys are doing this as volunteers and to thank the heavens that SODP exists, but it's never been clear to me how it all works and where you guys get your specific Sony-specific experience from.

it would be very helpful for potential donors and contributors to know this, not least because it would set up the proper expectations. some kind of public "current priorities list" or "WIP list" for open issues and devices would be helpful, too.

thanks!

**Comment by MarijnS95** [comment]
Simply put we have more than Sony's blessing to run this project: unlockable devices, official guides, and even proprietary support - it is their official project after all. But there is not enough manpower provided to support 20 devices on 8 platforms. That's where the tiny yet fast-paced group of _free time contributors_ comes in. We are pretty much supporting and maintaining all open source components (HALs, kernel, configuration and our own inventions) and keeping the entire thing up with the latest Android release. Mostly for our own use and the "greater good of open source software". This job becomes harder and harder with more stuff being pulled into proprietary land every release. We have to get creative, which ultimately defines this project and differentiates it from a stock release: as much opensourced software as possible.

**Comment by AngryGami** [comment]
>  This job becomes harder and harder with more stuff being pulled into proprietary land every release. 

So you are saying that Sony doesn't support you with source that they produce but instead closing in their implementation forcing you to redo what already had been done... This is grim and very unproductive.

**Comment by MarijnS95** [comment]
Not only Sony, the entire ecosystem surrounding phones and their ever-increasing suite of new hardware and features.

**Comment by kholk** [comment]
Sony gives us their open source archives, the ones that you can download from their website.
We do the rest on Open Devices in the most OSS fashion that we can, on our own: the help
that we get from Sony *which is what makes this entire thing possible* is proprietary code
releases that are "special" to Open Devices.

The stock team has literally nothing to do with us and we have literally nothing to do with
the stock team. We are two different "things", completely.

Besides that, apart from the Software Binaries release from Sony, this is basically a
totally community-driven project, done in our free time, aimed to give a solid
opensource base for Sony devices on Google's Android Open Source Project.

P.S.: We're basically a small team, if you look at the amount of supported devices... so,
if you want to help us by contributing, you're totally welcome!

**Comment by AngryGami** [comment]
@kholk 
>  so, if you want to help us by contributing, you're totally welcome!

I'm not qualified enough to help you with hardware and kernel stuff. I'm not a C/C++ programmer. Also none of potential contributors have access to Sony proprietary support as you do and this is another part of the issue.
Are there any chance we can get stock team involved here at least as consultants/experts?
I also curious how you do your build so that you don't get errors like above (assuming you don't) and why commits that break build end up in public branch?

**Comment by jerpelea** [comment]
@AngryGami 
We are offering the vendor part built for this purposem and we help when there are issues with the vendor or support during development but as Kholk stated this is a project built by community for community

**Comment by AngryGami** [comment]
@jerpelea I'm not sure which of my question you are answering to :)

**Comment by jerpelea** [comment]
> @jerpelea I'm not sure which of my questions you are answering to :)

your last message :)

**Comment by AngryGami** [comment]
> Also none of potential contributors have access to Sony proprietary support as you do and this is another part of the issue.

and your answer is:

> We are offering the vendor part built for this purposem and we help when there are issues with the vendor or support during development

Am I correct?
Who are "we" then? What does "vendor part built" means?

**Comment by jerpelea** [comment]
We in this context means Sony developers

Vendor binariyes means the odm file that you are download from our website and you should flash it on your device.

**Comment by kholk** [comment]
@AngryGami Regarding "broken commits", these are unexpected if we talk about ported devices *of course*, but they are totally EXPECTED when we are in a early porting phase: we tend to push whatever we have done that is totally correct for its own context on the official repositories in order to give access to this to any other developer that is potentially interested in it.

Mainly, we do this for two reasons:
1. We are "advertising" the fact that we are including this device in the Open Devices project
2. We are constantly seeking for help and we don't want developers to put effort in a component or a part of the work that we have already done. 

Regarding the access to proprietary code, I have to say UNFORTUNATELY, only Sony employees are allowed to view, modify, compile and release these bits: "external", community people have no access to that, even if someone of us have all the required knowledge to view, modify, help to fix, compile and test these bits. This is like that for legal reasons.
So no, during the porting job that I do, I don't see, compile nor fix any proprietary code: for this, I kindly ask Sony to fix the (potential) issue that I've found and they nicely work with us, community members, to provide in synchrony the "good stuff".

**Comment by chris42** [comment]
Maybe to help understand this: Alin @jerpelea is a Sony employee who is the liaison between Sony and the community project. That might make the "we" a bit clearer. :-)

To help you do not need to be a C/C++ programmer. There is a lot of other work to do as well (e.g. configurations, testing, merging patches).

**Comment by MarijnS95** [comment]
> Also none of potential contributors have access to Sony proprietary support as you do and this is another part of the issue.

Perhaps I made this seem like a way bigger issue than it really is. In fact we are in a "luxury" position with official support from Sony: we can ask questions and collaborate to push this Open Devices project forward without contributors ever needing to see a single line of proprietary code nor documentation. Most of the work is out in the open anyway: (Android) device tree configurations for features (`device-sony-` repositories) and sepolicy, the kernel, QualComm HALs from CodeAuroraForums, and our OSS-brew projects that aim to provide new functionality or replace a closed-source component, most of which live under `vendor/oss/`.

**Comment by jerpelea** [comment]
All needed patches were merged and now bug can be closed
